frontend/src/utils/generate-warnings.ts

```diff
@@ -676,13 +689,64 @@ function produceSpecificCourseWarnings(
   warnings = warnings.concat(
     checkPrerequisites(term.classes, tracker, term.termId)
   );
+  ///// ADD CODE HERE TO CHECK IF THERE ARE DUPLICATES
+  warnings = warnings.concat(
+    checkDuplicates(term.classes, tracker, term.termId)
+  );
+  ////////////// END ///////////
+  return warnings;
+}
+
+/**
+ * Checks that the courses are not duplicates.
+ * @param toCheck The schedule to check.
+ * @param tracker tracker for courses taken
+ * @param termId the id for this term for the warning
+ */
+function checkDuplicates(
+  toCheck: ScheduleCourse[],
+  tracker: CourseTakenTracker,
+  termId: number
+): CourseWarning[] {
+  const warnings: CourseWarning[] = [];
+  for (let i = 0; i < toCheck.length; i++) {
+    const course = toCheck[i];
+    if (
+      courseCode(course) != "XXXX9999" &&
+      tracker.contains(courseCode(course))
+    ) {
+      warnings.push({
+        subject: course.subject,
+        classId: course.classId,
+        message: `${courseCode(course)}: is in your schedule multiple times`,
+        termId: termId,
+      });
+      warnings.push({
+        subject: course.subject,
+        classId: course.classId,
+        message: `${courseCode(course)}: is in your schedule multiple times`,
+        termId: tracker.getTermId(courseCode(course)),
```

Comment:
I was curious as to why you add two warnings for the same course here. Wouldn't a single warning be sufficient.

Reply:
So the reason for this is because the "first" of the duplicates does not think it is a duplicate, because it only sees course in the schedule before this class, so when the second class notices the first as a duplicate, it needs to mark itself and the other one as a duplicate. I just based it on the structure of the current code because right now the tracker only knows the course before the term you're on.